qi-lib/switch.rkt

```diff
@@ -19,28 +22,21 @@
        (switch clause ...))])
 
 (define-syntax-parser switch-lambda
-  [(_ (arg:id ...) expr:expr ...)
-   #'(lambda (arg ...)
-       (switch (arg ...)
-         expr ...))]
-  [(_ rest-args:id expr:expr ...)
-   #'(lambda rest-args
-       (switch (rest-args)
+  [(_ args:formals expr:expr ...)
+   #:with ags (params-parser #'args)
+   #'(lambda args
+       (switch ags
```

Comment:
nit: might be worth adding a comment here that the parsed `ags` is being passed to `switch` while the unparsed `args` are passed to `lambda`, so that lambda can bind keyword arguments in scope while the flow itself does not receive them directly. Otherwise it may not be super obvious to future devs (like ourselves 😄 ) why these are different.